phases/17-infrastructure-and-production/01-managed-llm-platforms/code/main.py

```
@@ -0,0 +1,110 @@
+    print("=" * 80)
+    tokens_per_day = 5_000_000
+    primary_cost = (tokens_per_day / 1e6) * 10.00
+    gateway_overhead_pct = 3.0
+    failover_headroom_pct = 10.0
+    uplift = primary_cost * (gateway_overhead_pct + failover_headroom_pct) / 100
+    print(f"Primary daily spend: ${primary_cost:.2f}")
+    print(f"Gateway overhead ({gateway_overhead_pct:.0f}%): ${primary_cost * gateway_overhead_pct / 100:.2f}/day")
+    print(f"Idle secondary headroom ({failover_headroom_pct:.0f}%): ${primary_cost * failover_headroom_pct / 100:.2f}/day")
+    print(f"Total uplift: ${uplift:.2f}/day")
+    print(f"Monthly uplift: ${uplift * 30:.2f}")
+    print(f"Cost of one multi-hour regional outage without redundancy: customer churn, SLA credits, war-room time")
```

Ruff (0.15.11)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)